### PR TITLE
Immovable Rod changes

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR

A few changes & fixes for immovable rods:

1. Prevents rods from killing/deleting their polymoprhed paused parent entity
2. Separates timed despawn from base Immovable Rods
3. Adds a way for rods to just damage mobs instead of gibbing & deleting (for wizard rod form)
4. Adds wizard rod entity

## Why / Balance

When Wizards are added this will help prevent some ire so the crew doesn't get deleted and so the wizard doesn't kill themselves after transforming.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by deltanedas** [comment]
i knew from first line livrah wizard rod real....

**Comment by UbaserB** [comment]
does this fix include stopping the game from crashing whenether it hits you (along with not spawning anything iirc)?

**Comment by keronshb** [comment]
> does this fix include stopping the game from crashing whenether it hits you (along with not spawning anything iirc)?

I haven't had any crashes with gibbing when testing. But if it's causing issues you could just turn the base rod/despawn rod to gib = false and it'll still delete walls/tiles but not gib people.

**Comment by UbaserB** [comment]
> > does this fix include stopping the game from crashing whenether it hits you (along with not spawning anything iirc)?
> 
> I haven't had any crashes with gibbing when testing. But if it's causing issues you could just turn the base rod/despawn rod to gib = false and it'll still delete walls/tiles but not gib people.

it was just the other day i got hit by a rod and me along with like 2 other people crashed.

image

**Comment by keronshb** [comment]
> > > does this fix include stopping the game from crashing whenether it hits you (along with not spawning anything iirc)?
> > 
> > 
> > I haven't had any crashes with gibbing when testing. But if it's causing issues you could just turn the base rod/despawn rod to gib = false and it'll still delete walls/tiles but not gib people.
> 
> it was just the other day i got hit by a rod and me along with like 2 other people crashed (would it be better to say game entirely frozen with no way to fix unless restarting?)
> 
> image

OH I thought I had gotten that from deleting in a paused map. I might just disable gibbing on rods for now.

**Comment by metalgearsloth** [comment]
> > > does this fix include stopping the game from crashing whenether it hits you (along with not spawning anything iirc)?
> > 
> > 
> > I haven't had any crashes with gibbing when testing. But if it's causing issues you could just turn the base rod/despawn rod to gib = false and it'll still delete walls/tiles but not gib people.
> 
> it was just the other day i got hit by a rod and me along with like 2 other people crashed (would it be better to say game entirely frozen with no way to fix unless restarting?)

Link the github issue.

**Comment by UbaserB** [comment]
#22625 #26366

**Comment by keronshb** [comment]
> #22625 #26366

Looks like this is happening with gibbing and it's not exclusive to the rod.

**Comment by UbaserB** [comment]
> Looks like this is happening with gibbing and it's not exclusive to the rod.

image

**Comment by keronshb** [comment]
> > Looks like this is happening with gibbing and it's not exclusive to the rod.
> 
> image

Happens with Smite too.

Some things that are gibbing are not dropping organs, which prevents a brain from dropping so the mind doesn't know where to go.

File: Content.Server/ImmovableRod/ImmovableRodSystem.cs [hunk 1]
```diff
@@ -94,12 +97,28 @@ private void OnCollide(EntityUid uid, ImmovableRodComponent component, ref Start
...
         if (TryComp<BodyComponent>(ent, out var body))
         {
             component.MobCount++;
-
             _popup.PopupEntity(Loc.GetString("immovable-rod-penetrated-mob", ("rod", uid), ("mob", ent)), uid, PopupType.LargeCaution);
+
+            if (!component.ShouldGib)
+            {
+                if (!TryComp<DamageableComponent>(ent, out var damageable) || component.Damage == null)
+                    return;
```
Comment (metalgearsloth) [inline Content.Server/ImmovableRod/ImmovableRodSystem.cs:116]: Checking null is significantly faster than trycomp so you should reverse this order.

Comment (keronshb) [reply Content.Server/ImmovableRod/ImmovableRodSystem.cs]: flipped

**Pull request merged**